src/pyedb/dotnet/edb_core/components.py

```diff
@@ -848,12 +848,14 @@ def create_port_on_pins(self, refdes, pins, reference_pins, impedance=50.0, port
         if len([pin for pin in pins if isinstance(pin, str)]) == len(pins):
             cmp_pins = []
             for pin_name in pins:
-                cmp_pin = [pin for pin in list(refdes_pins.values()) if pin_name == pin.name]
-                if not cmp_pin:
-                    cmp_pin = [pin for pin in list(refdes_pins.values()) if pin_name == pin.name.split("-")[1]]
-                if cmp_pin:
-                    cmp_pins.append(cmp_pin[0])
+                cmp_pins = [pin for pin in list(refdes_pins.values()) if pin_name == pin.name]
+                if not cmp_pins:
+                    for pin in list(refdes_pins.values()):
+                        if pin.name and "-" in pin.name:
+                            if pin_name == pin.name.split("-")[1]:
+                                cmp_pins.append(pin)
             if not cmp_pins:
+                self._logger.warning("No pin found for creating port, skipping.")
```

Comment:
```suggestion
                self._logger.warning("No pin found during port creation. Port is not defined.")
```
I like the second message warning more than the first :)